src/snowplow/prospect/prospectEventHandler.ts

```diff
@@ -0,0 +1,115 @@
+import { buildSelfDescribingEvent } from '@snowplow/node-tracker';
+import { SelfDescribingJson } from '@snowplow/tracker-core';
+import {
+  ObjectUpdate,
+  ProspectEventPayloadSnowplow,
+  ProspectReviewStatus,
+  SnowplowEventMap,
+} from './types';
+import { config } from '../../config';
+import { EventHandler } from '../EventHandler';
+import { getTracker } from '../tracker';
+
+type ObjectUpdateEvent = Omit<SelfDescribingJson, 'data'> & {
+  data: ObjectUpdate;
+};
+
+type ProspectContext = Omit<SelfDescribingJson, 'data'> & {
+  data: {
+    object_version: string;
+    prospect_id: string;
+    url?: string;
```

Comment:
@jpetto, it will be, alongside with most other properties marked as optional here - but here the shape of the data follows the Snowplow schema so that's where the URL is optional but `prospect_id` is not.